### PR TITLE
fix(DefaultEvictor): softIdleTimeoutMillis handle -1

### LINKED ISSUE #192: softIdleTimeoutMillis at -1 (default) does not work
The README claims that softIdleTimeoutMillis has a default of -1 which means no eviction would occur. However, if I write a simple test case I do see eviction happening:
```
const genericPool = require("generic-pool")

const factory = {
    create: async () => {
        console.log('creating resource');
        return new Object();
    },
    destroy: async () => console.log('destroying resource'),
}
const opts = {
    evictionRunIntervalMillis: 100
}
var myPool = genericPool.createPool(factory, opts);
const a = await myPool.acquire();
console.log('got first resource');
await myPool.release(a);
```

with output:
```
creating resource
got first resource
destroying resource <<< this should not happen on softTimeout (because -1) and hardTimeout is only after 30 secs while this logging appears after 100msecs (first eviction run)
```

Looking at `DefaultEvictor.js` I see no special handling for the `-1` value:
```
    if (config.softIdleTimeoutMillis < idleTime && config.min < availableObjectsCount) {
      return true
    }

    if (config.idleTimeoutMillis < idleTime) {
      return true
    }

    return false
```

Berhaps a better default for softIdleTimeoutMillis would be `Infinity` as that would mean no eviction, or we should add special handling for `-1` in de DefaultEvictor

### PR DESCRIPTION
**Pull request by DiegoRBaquero**
fixes #192

### CODE REVIEW

File: lib/DefaultEvictor.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ class DefaultEvictor {
     const idleTime = Date.now() - pooledResource.lastIdleTime;
 
     if (
+      config.softIdleTimeoutMillis > 0
```
Comment (sandfox) [inline lib/DefaultEvictor.js]:
```suggestion
      config.softIdleTimeoutMillis > 0 &&
```

**Comment by DiegoRBaquero** [comment]
My bad, did it through GitHub haha! Thanks for the fix

**Comment by sandfox** [comment]
no worries, thanks for doing the work in the first place! This is now released as v3.5.0

**Comment by sandfox** [comment]
merged in via #245

**Pull request closed without merging**